test/unit/config_test.rb

```diff
@@ -0,0 +1,417 @@
+        final_options['color'].must_equal true
+        final_options['create_lockfile'].must_equal true
+        final_options['backend_cache'].must_equal true
+      end
+    end
+
+    describe 'when the shell command is used' do
+      let(:command) { :shell }
+      it 'should have the correct defaults' do
+        expected = ['reporter', 'type'].sort
+        seen_fields.must_equal expected
+        final_options['reporter'].must_be_kind_of Hash
+        final_options['reporter'].count.must_equal 1
+        final_options['reporter'].keys.must_include 'cli'
+      end
+    end
+  end
+
+  # ========================================================================== #
+  #                            Reading CLI Options
+  # ========================================================================== #
+  # The config facility supports passing in CLI options in the constructor, so
+  # that it can handle merging internally. That is tested here.
+  #
+  #  This is different than storing options
+  # in the config file with the same name as the CLI options, which is
+  # tested under 'CLI Options Stored in File'
+  describe 'reading CLI options' do
+    let(:cfg) { Inspec::Config.new(cli_opts) }
+    let(:final_options) { cfg.final_options }
+    let(:seen_fields) { cfg.final_options.keys.sort }
+
+    describe 'when the CLI opts are present' do
+      let(:cli_opts) do
+        {
+          color: true,
+          'string_key' => 'string_value',
+          array_value: [1,2,3],
+        }
+      end
+
+      it 'should transparently round-trip the options' do
+        expected = ['color', 'array_value', 'reporter', 'string_key', 'type'].sort
+        seen_fields.must_equal expected
+        final_options[:color].must_equal true
+        final_options['color'].must_equal true
+        final_options['string_key'].must_equal 'string_value'
+        final_options[:string_key].must_equal 'string_value'
+        final_options['array_value'].must_equal [1,2,3]
+        final_options[:array_value].must_equal [1,2,3]
+      end
+    end
+  end
+
+  # ========================================================================== #
+  #                          CLI Options Stored in File
+  # ========================================================================== #
+  describe 'reading CLI options stored in the config file' do
+    let(:cfg) { Inspec::Config.new({}, cfg_io) }
+    let(:final_options) { cfg.final_options }
+    let(:cfg_io) { StringIO.new(ConfigTestHelper.fixture(fixture_name)) }
+    let(:seen_fields) { cfg.final_options.keys.sort }
+
+    # These two test cases have the same options but in different file versions.
+    describe 'when the CLI opts are present in a 1.1 file' do
+      let(:fixture_name) { :like_legacy }
+      it 'should read the options' do
+        expected = ['color', 'reporter', 'target_id', 'type'].sort
+        seen_fields.must_equal expected
+        final_options['color'].must_equal "true"  # Dubious
```

Comment:
Dubious how so? Could you elaborate in the comment?

Reply:
When we read a JSON file, should we expect the String "true" to be returned as a String, or as `true`, an instance of TrueClass?

Reply:
Ah, could we elaborate in the comment?